### PR TITLE
feat: fetch metrics for the dynamo and elasticache resources

### PR DESCRIPTION
**Pull request by nand4011**
Change the get metadata functions for dynamo and elasticache to return a resource that will later be serialized into json.

Add code for fetching metrics based on the dynamo and elasticache resources.

Change the linter cli to print out the full json including the metrics.

### CODE REVIEW

File: momento/src/commands/cloud_linter/dynamodb.rs [hunk 1]
```diff
@@ -1,47 +1,184 @@
...
+        "Sum" => &[
+            "ConsumedReadCapacityUnits",
+            "ConsumedWriteCapacityUnits",
+            "ReadThrottleEvents",
+            "WriteThrottleEvents",
+            "TimeToLiveDeletedItemCount",
+            "TransactionConflict",
+            "ConditionalCheckFailedRequests",
+        ],
+        "Average" => &[
```
Comment (bruuuuuuuce) [inline momento/src/commands/cloud_linter/dynamodb.rs:26]: I think we also need the average of 

```
ConsumedReadCapacityUnits
ConsumedWriteCapacityUnits
```

I think I forgot to update the doc for these, I will do that now and make sure its updated

File: momento/src/commands/cloud_linter/dynamodb.rs [hunk 2]
```diff
@@ -1,47 +1,184 @@
...
+};
+
+const DDB_GSI_METRICS: Map<&'static str, &'static [&'static str]> = phf_map! {
+    "Sum" => &[
+            "ConsumedReadCapacityUnits",
+            "ConsumedWriteCapacityUnits",
+            "ReadThrottleEvents",
+            "WriteThrottleEvents",
+        ],
+    "Average" => &[
```
Comment (bruuuuuuuce) [inline momento/src/commands/cloud_linter/dynamodb.rs:47]: same here

File: momento/src/commands/cloud_linter/metrics.rs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    async fn append_metrics(
+        &mut self,
+        metrics_client: &Client,
+        limiter: Arc<DefaultDirectRateLimiter>,
+    ) -> Result<(), CliError> {
+        let metric_target = self.create_metric_target()?;
+        let metrics =
+            query_metrics_for_target(&metrics_client, Arc::clone(&limiter), metric_target).await?;
+        self.set_metrics(metrics);
+        self.set_metric_period_seconds(60 * 60 * 24);
```
Comment (bruuuuuuuce) [inline momento/src/commands/cloud_linter/metrics.rs:60]: This is 1 day right?

**Review by bruuuuuuuce — COMMENTED** [review]
Looks good! Left a few clarifying questions/nits

File: momento/src/commands/cloud_linter/metrics.rs (hunk 3 above)
Comment (nand4011) [reply momento/src/commands/cloud_linter/metrics.rs:60]: It is.

File: momento/src/commands/cloud_linter/dynamodb.rs (hunk 1 above)
Comment (bruuuuuuuce) [reply momento/src/commands/cloud_linter/dynamodb.rs:24]: @nand4011 we want both the sum and the average for these metrics

**Review by bruuuuuuuce — APPROVED** [review]
:shipit:

**Pull request merged**